src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,63 @@
 import React from 'react';
 import './CurrentUser.scss';
+import { loadUser } from '../../api';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+type Props = {
+  userId: number;
+  userClear: () => void;
+};
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+interface State {
+  user: User | null;
+}
+
+export class CurrentUser extends React.Component<Props, State> {
+  state: State = {
+    user: null,
+  };
+
+  async componentDidMount() {
+    this.reloadUser(this.props.userId);
+  }
+
+  async componentDidUpdate(prevProps: Props) {
+    if (prevProps.userId !== this.props.userId) {
+      this.reloadUser(this.props.userId);
+    }
+  }
+
+  async reloadUser(userId: number) {
+    const user = await loadUser(userId);
+
+    this.setState({ user });
+  }
```

Comment:
```suggestion
  componentDidMount() {
    this.reloadUser();
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.userId !== this.props.userId) {
      this.reloadUser();
    }
  }

  async reloadUser(userId: number) {
    const user = await loadUser(this.props.userId);

    this.setState({ user });
  }
```
no need async on lifecycle methods here